src/classes/RD2_RecurringDonations_TDTM.cls

```diff
@@ -260,92 +260,136 @@ public class RD2_RecurringDonations_TDTM extends TDTM_Runnable {
             return;
         }
 
-        autoPopulateAccounts(rds);
+        DataRegulator dataRegulator = new DataRegulator(currentDate);        
+        dataRegulator.adjust(rds, oldRds);
+    }
 
-        for (npe03__Recurring_Donation__c rd : rds) {
-            if (rd.npe03__Installment_Period__c != RD2_Constants.INSTALLMENT_PERIOD_MONTHLY && rd.Day_of_Month__c != null) {
-                rd.Day_of_Month__c = null;
-            }
-            //map the Day of Month value of "31" to the Last_Day value
-            if (rd.Day_of_Month__c == '31') {
-                rd.Day_of_Month__c = RD2_Constants.DAY_OF_MONTH_LAST_DAY;
-            }
 
-            //set the default installment frequency when value is not provided on the RD insert
-            if (rd.InstallmentFrequency__c == null && triggerAction == TDTM_Runnable.Action.BeforeInsert) {
-                rd.InstallmentFrequency__c = RD2_Constants.DEFAULT_INSTALLMENT_FREQUENCY;
-            }
+    /***
```

Comment:
Would would you think of moving this to a separate class (`RD2_DataRegulatorService`?)? It feels like this behavior isn't necessarily part of TDTM specifically, but instead a service used by TDTM and others.